### ISSUE TITLE
The Client connection port change from time to time.

### ISSUE THREAD

**Issue by acaibird** [eclipse/californium#812]
We found that after NB-IOT sensor hiberbates, the base station will reassign an external network port to it. So server should update the registration information of the sensor.  My solution is as follows:
```
        Registration registration = server.getRegistrationService().getByEndpoint(endpointName);
		// the registration information saved by the server
		Identity identity = registration.getIdentity();
		InetSocketAddress isa = identity.getPeerAddress();
                //if the client's address or port change, update the client's registration information
		if (isa.getAddress() != response.getSource()
                                                  || isa.getPort() != response.getSourcePort()) {
			InetSocketAddress tem = new InetSocketAddress(response.getSource(),
					response.getSourcePort());
                      /**********we modify the source code*********/
			registration.getIdentity().setInetSocketAddress(tem);
		}
```
As you can see，our method is very bad, we check the registration information of the client every time it uploads data.
I want to ask if you have a better solution. Thanks.

**Comment by boaks** [comment]
Look's similar to LWM2M, do you use leshan?

And sure, if your "client" changes the address:port and you want to send a request to the "client" (which is then changing into a coap-server), you must adapt the address.
(It gets much harder, if you use DTLS.)

**Comment by acaibird** [comment]
Yes! I use leshan. In this scenario, changes in connection information are tricky.

**Comment by acaibird** [comment]
The sensor maintains a long connection and sends the heartbeat packet regularly, so that the connection information will not change. Or the server saves the command until the sensor sends a request and then sends it. The first method consumes a lot of power, and the second method cannot send commands in real time.

**Comment by boaks** [comment]
The second one is named "queue mode".
Yes generally, if your device sleeps or doesn't communicate for a period of time it gets "unreachable". It's unreachable at the IP-layer and therefore neither coap nor lwm2m could change that "break".

**Comment by acaibird** [comment]
If I use UDP broadcasting mode, can I solve this problem? Sure, server and client in same broadcast network.

**Comment by boaks** [comment]
> NB-IOT sensor hiberbates

If the hiberbated NB-IOT sensor can receive multicast, try it :-).

I assume, it doesn't work, but try it. The current californium 2.0.x HEAD includes support for coap multecast, see cf-helloworld-client MulticastTestClient.
